src/binary_tree/avl.ts

```diff
@@ -0,0 +1,250 @@
+import { AVLNode } from "./avl_node";
+import { IAVLNode, IAVLTraverseCallback, IAVLTree } from "./interfaces";
+
+
+export class AVL implements IAVLTree {
+  root: IAVLNode | null;
+
+  constructor() {
+    this.root = null;
+  }
+
+  traversePreOrder(node: IAVLNode | null, callback: IAVLTraverseCallback) {
```

Comment:
All trees have exactly the same implementation of traversal methods. We must have some specific class that release these methods, and all trees simply inherit from it.